_posts/2020-10-23-deconvolution.md

```diff
@@ -0,0 +1,347 @@
+---
+layout: post
+title: Image Analysis Redux
+tagline: Dask + CuPy + RL
+author: John Kirkham (NVIDIA) and Ben Zaitlen (NVIDIA)
+theme: twitter
+---
+{% include JB/setup %}
+
+Summary
+-------
+
+
+Image Analysis Redux
+--------------------
+
+[Last year](https://blog.dask.org/2019/08/09/image-itk) we experimented with Dask/ITK/Scikit-Image
+to perform large scale image analysis on a stack of 3D images.  Specifically, we looked at `deconvolution`
+a common method to _deblur_ images.  Now, a year later, we return to these experiments with a better
+understanding of how Dask and CuPy can interact, enhanced serialization methods, and support from the
+open-source community.
+
+Previously we used the [Richardson Lucy (RL)](https://en.wikipedia.org/wiki/Richardson%E2%80%93Lucy_deconvolution) deconvolution
+algorithm from ITK and [Scikit-Image](https://github.com/scikit-image/scikit-image/blob/master/skimage/restoration/deconvolution.py#L329).
+We left off at theorizing how GPUs could potential help accelerate these workflows.  Starting with Scikit-Image's
+implementation, we naively tried replacing `scipy.signal.convolve` calls with `cupyx.scipy.ndimage.convolve`, and while
+performance improved, it did not improve _significantly_ -- that is, we did not get the 100X speed we were looking for
+
+Often, in working with images, we hit performance roadblocks in real space where data represents position.  When this happens,
+we can reach for a common tool, the [Fast-Fourirer Transform FFT](https://en.wikipedia.org/wiki/Fast_Fourier_transform).  FFTs convert
+the data to from real space to frequency space where data now represents the rate of change of intensity between pixels.  This conversion
+is extremely fast on both CPUs and GPUs and the algorithm we can write with FFTs similarly are accelerated.
+And while it is challenging to write an equivalent algorithm in FFT space (thank you mathematicians!)
+the cost of transformation + the cost of the algorithm is still significantly faster than performing the original algorithm in real space.
```

Comment:
A bit of a mixed metaphor here. "the *cost* of ... is significantly *faster*".

Cost should be less / lower, or the speed should be faster.

Reply:
Thanks @TomAugspurger for the early review.  I'll fix these up and some other thoughts shortly

Reply:
Switched to "lower" in recent changes